src/v/cluster/tm_stm.h

```diff
@@ -266,4 +324,14 @@ class tm_stm final : public persisted_stm<> {
     serialize_hosted_transactions(tm_tx_hosted_transactions hr);
 };
 
+inline txlock_unit::~txlock_unit() noexcept {
+    if (_stm) {
+        _units->return_all();
```

Comment:
there is already a check in `return_all` for `_stm` pointer

Reply:
never mind I've done it